src/js/media-captions-list.js

```diff
@@ -0,0 +1,119 @@
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+import {
+  MediaUIEvents,
+  MediaUIAttributes,
+  TextTrackKinds,
+} from './constants.js';
+import { parseTextTrackStr, splitTextTracksStr } from './utils/captions.js';
+import MediaChromeListbox from './media-chrome-listbox.js';
+
+const CC_PREFIX = '!cc';
+const isCC = (extendedTextTrackStr = '') =>
+  extendedTextTrackStr?.startsWith?.(CC_PREFIX);
+const withoutCC = (extendedTextTrackStr = '') => {
+  if (!isCC(extendedTextTrackStr)) return extendedTextTrackStr;
+  return extendedTextTrackStr.slice(CC_PREFIX.length);
+};
+
+const DEFAULT_UNKNOWN_LABEL = 'UNKNOWN';
+const DEFAULT_CC_LABEL_PART = '(cc)';
+const DEFAULT_FORMATTER = (value) => {
+  const textTrackStr = withoutCC(value);
+  const baseLabel =
+    parseTextTrackStr(textTrackStr)?.label ?? DEFAULT_UNKNOWN_LABEL;
+  const suffix = isCC(value) ? ` ${DEFAULT_CC_LABEL_PART}` : '';
+  return `${baseLabel}${suffix}`;
+};
+const parseAsSubCCStr = (str, kind = TextTrackKinds.SUBTITLES) => {
+  if (!str) return [];
+  if (kind === TextTrackKinds.SUBTITLES) return splitTextTracksStr(str);
+  return splitTextTracksStr(str).map(
+    (textTrackStr) => `${CC_PREFIX}:${textTrackStr}`
+  );
+};
+
+class MediaCaptionsList extends MediaChromeListbox {
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  constructor({ format = DEFAULT_FORMATTER } = {}) {
+    super({ format });
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (
+      [
+        MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+        MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      ].includes(attrName) &&
+      oldValue !== newValue
+    ) {
+      this.data = [
```

Comment:
given that this data is treated separately in the media controller state management architecture, we need some way to "normalize" to a single list (on the assumption we want our list component to be generic).